### DISCUSSION TITLE
2.16.1: Handle possibility of configs folder not existing

### DISCUSSION THREAD

**Discussion by divinewind420** [ThioJoe/YT-Spammer-Purge#684] [Ideas]
2.16.1 will assume the configs folder exists, so you cannot create additional config files, if folder does not exist, get error instead. Will generate additional config files if you manually create the folder.
Suggest adding config folder exists? options.
if config folder does not exist: Ask/Autocreate/abort

**Comment by ThioJoe** [comment]
Hm it shouldn't do that, must have been an oversight. I'll have to fix that.

**Reply by Lagicrus** [reply]
Had the same issue, on upgrade from 2.15.1 it hard crashed due to the `./config` not existing

**Comment by ThioJoe** [comment]
Ok should be fixed with the new 2.16.2 release

**Reply by Lagicrus** [reply]
Seems to still show a bug with config

<details>
<summary>Bug Z-3</summary>
Getting info about latest updates...
------------------------------------------------------------------------------------------
 A new version is available! Visit TJoe.io/latest to see what's new.
   > Current Version: 2.15.1
   > Latest Version: 2.16.2
------------------------------------------------------------------------------------------

Update Now? (y/n): y

> Downloading version: 2.16.2

> Extracting...
> Installing...
Traceback (most recent call last):
  File "/usr/lib/python3.8/shutil.py", line 791, in move
    os.rename(src, real_dst)
FileNotFoundError: [Errno 2] No such file or directory: './config' -> '/home/ocean/YT-Spammer-Purge-2.15.1/temp/YT-Spammer-Purge-2.16.2/config'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "YTSpammerPurge.py", line 1611, in <module>
    main()
  File "YTSpammerPurge.py", line 1588, in main
    continueRunning = primaryInstance(miscData)
  File "YTSpammerPurge.py", line 875, in primaryInstance
    files.check_for_update(version, updateReleaseChannel)
  File "/home/ocean/YT-Spammer-Purge-2.15.1/Scripts/files.py", line 350, in check_for_update
    move("./config", extraFolderPath)
  File "/usr/lib/python3.8/shutil.py", line 811, in move
    copy_function(src, real_dst)
  File "/usr/lib/python3.8/shutil.py", line 435, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.8/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: './config'
------------------------------------------------
Error Message: [Errno 2] No such file or directory: './config'

Unknown Error - Code: Z-3 occurred. If this keeps happening, consider posting a bug report
on the GitHub issues page, and include the above error info.
Short Link: TJoe.io/bug-report
</details>

**Comment by ThioJoe** [comment]
Moving to issues. This is actually not related to the original error, but rather the linux update function.

**Reply by Firecul** [reply]
> This is actually not related to the original error, but rather the linux update function

That's why I just pull from git, I've not had any issues as long as I stayed to the newest tag.
